src/librustc_driver/lib.rs

```diff
@@ -327,8 +331,13 @@ pub fn run_compiler(
             return sess.compile_status();
         }
 
+        let expanded_ast = compiler.expansion()?;
```

Comment:
This call and the callback should be moved above `compiler.prepare_outputs()?;` line since `expansion` occurs before `prepare_outputs`.

I'd also just drop the `expanded_ast` variable and keep the existing `compiler.expansion()?` code.